### ISSUE TITLE
security flaw when editOnly is set to true?

### ISSUE THREAD

**Issue by HamzaKhait** [ether/etherpad-lite#3483]
Hello,

I've noticed that users can create pads even if editOnly option is set to true in settings, here's the scenario : 
- user opens https://example.com/etherpad
- user sets pad name and clicks Ok
- Etherpad shows "You do not have permission to access this pad"
- user refreshes the web page (https://example.com/etherpad/p/PadName)
- Etherpad creates and shows the pad

That means that when etherpad show the permission message, it creates the pad, which is not normal because I only allow pads creation through etherpad HTTP API.

Any help to fix this please ?

**Comment by muxator** [comment]
Hi @HamzaKhait.

This issue sounds ugly, but I am not being able to trigger it.

I am using the Etherpad 1.7.0 with the following configuration (I am including other options that may be different from yours in order to ensure we are aligned):

```javascript
{
  /*
   * If this option is enabled, a user must have a session to access pads.
   * This effectively allows only group pads to be accessed.
   */
  "requireSession" : false,

  /*
   * Users may edit pads but not create new ones.
   *
   * Pad creation is only via the API.
   * This applies both to group pads and regular pads.
   */
  "editOnly" : true,

  /*
   * If set to true, those users who have a valid session will automatically be
   * granted access to password protected pads.
   */
  "sessionNoPassword" : false,

  /*
   * This setting is used if you require authentication of all users.
   *
   * Note: "/admin" always requires authentication.
   */
  "requireAuthentication" : false,

  /*
   * Require authorization by a module, or a user with is_admin set, see below.
   */
  "requireAuthorization" : false,
}
```

Also: you are using a reverse proxy, aren't you?
That is another difference with my setup, albeit I do not think it could be relevant.

**Comment by HamzaKhait** [comment]
Hello,

I am using etherpad version 1.6.6 behind nginx reverse proxy and I have the same configuration 

```
  /* Users must have a session to access pads. This effectively allows only group pads to be accessed. */
  "requireSession" : false,

  /* Users may edit pads but not create new ones. Pad creation is only via the API. This applies both to group pads and regular pads. */
  "editOnly" : true,

  /* Users, who have a valid session, automatically get granted access to password protected pads */
  "sessionNoPassword" : false,

  /* if true, all css & js will be minified before sending to the client. This will improve the loading performance massivly,
     but makes it impossible to debug the javascript/css */
  "minify" : true,

  /* How long may clients use served javascript code (in seconds)? Without versioning this
     may cause problems during deployment. Set to 0 to disable caching */
  "maxAge" : 21600, // 60 * 60 * 6 = 6 hours

  /* This is the absolute path to the Abiword executable. Setting it to null, disables abiword.
     Abiword is needed to advanced import/export features of pads*/
  "abiword" : null,

  /* This is the absolute path to the soffice executable. Setting it to null, disables LibreOffice exporting.
     LibreOffice can be used in lieu of Abiword to export pads */
  "soffice" : null,

  /* This is the path to the Tidy executable. Setting it to null, disables Tidy.
     Tidy is used to improve the quality of exported pads*/
  "tidyHtml" : null,

  /* Allow import of file types other than the supported types: txt, doc, docx, rtf, odt, html & htm */
  "allowUnknownFileEnds" : false,

  /* This setting is used if you require authentication of all users.
     Note: /admin always requires authentication. */
  "requireAuthentication" : false,

  /* Require authorization by a module, or a user with is_admin set, see below. */
  "requireAuthorization" : false,

  /*when you use NginX or another proxy/ load-balancer set this to true*/
  "trustProxy" : true,

```

**Comment by HamzaKhait** [comment]
This issue was fixed when I updated my etherpad to 1.7.5
